### ISSUE TITLE
[Dev] fix non-deterministic GEC output caused by BERT dropout at inference

### ISSUE THREAD

**Issue by RusCucumber** [RusCucumber/gector-ja#3]
## Task Description
`GEC.correct()` / `GEC.predict()` return different results for the exact same input text across repeated calls, because BERT's internal Dropout layers stay active even at inference time.

## Branch Type
- [x] `fix/`: Bug fix (.py)

## Implementation Details
Root cause: in `model.py`'s `get_model()`, the BERT encoder is connected with an explicit `training=bert_trainable` argument:

```python
embedding = encoder(input_ids, attention_mask=attention_mask, training=bert_trainable)[0]
```

Since `bert_trainable` defaults to `True` (and is never overridden in `main.py`), this bakes `training=True` into the Functional API graph at model-construction time. As a result, `predict()`'s call `self.model(input_ids, training=False)` no longer propagates `training=False` down into the BERT encoder, so Dropout keeps firing during inference and produces different `labels_probs`/`detect_probs` on every call, even for identical `input_ids`.

Verified with a reproduction script:
- Current code: two `predict()` calls on identical `input_ids` differ (max abs diff on `labels_probs` ≈ 5.7e-4, argmax labels differ).
- Removing the explicit `training=bert_trainable` from the `encoder(...)` call (letting Keras auto-propagate the outer `training` value) makes two calls bit-identical (diff = 0.0), while training-time behavior (`model.fit()` implicitly passing `training=True`) is unaffected.

Fix: change
```python
embedding = encoder(input_ids, attention_mask=attention_mask, training=bert_trainable)[0]
```
to
```python
embedding = encoder(input_ids, attention_mask=attention_mask)[0]
```

## Action Steps
- [ ] Create a branch named `fix/deterministic-inference-dropout-#<issue-number>`
- [ ] Implement the bug fix
- [ ] Add or update tests as necessary
- [ ] Ensure all tests pass locally
- [ ] Push the branch, open a PR, and review the code changes

## Additional Notes
Log evidence that surfaced the bug: identical input `多分一番合いますと思う。` produced two different corrections (`多分一番だと思う。` and `多分一番合うと思う。`) across separate `/correct` requests.
